doc/guides/beyond-GSG.rst

```diff
@@ -0,0 +1,277 @@
+.. _beyond-gsg:
+
+Beyond the Getting Started Guide
+################################
+
+The :ref:`getting_started` gives a straight-forward path to set up
+your Linux, macOS, or Windows enviromnent for Zephyr development. In
+this document, we delve deeper into Zephyr development setup
+issues and alternatives.
+
+.. _python-pip:
+
+Python and pip
+**************
+
+Python 3 and its package manager, pip\ [#pip]_, are used extensively by Zephyr
+to install and run scripts required to compile and run Zephyr
+applications, set up and maintain the Zephyr development environment,
+and build project documentation.
+
+Depending on your operating system, you may need to provide the
+``--user`` flag to the ``pip3`` command when installing new packages. This is
+documented throughout the instructions.
+See `Installing Packages`_ in the Python Packaging User Guide for more
+information about pip\ [#pip]_, including `information on -\\-user`_.
+
+- On Linux, make sure ``~/.local/bin`` is at the front of your :envvar:`PATH`
```

Comment:
The issue with this being in "beyond the GSG" is that this is required for many users in order to get a working build system. We already had complaints that this wasn't shown before the OS instructions, which is why we moved it to the top.

Reply:
But this part is in the GSG as well, and it is repeated as troubleshooting here - makes sense imo.